### PR TITLE
Introduction to sync JS problems

### PR DESCRIPTION
**Pull request by JulianPasquale**

### CODE REVIEW

File: src/slides/class4/slides/BlockingCode.jsx [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+      <Text fontSize={30}>
+        Podemos pensar en un thread (hilo) como un proceso que los programas utilizan para ejecutar tareas, y cada thread solo puede ejecutar una tarea a la vez. JavaScript ejecuta todo el código en un solo thread (<a href="https://developer.mozilla.org/en-US/docs/Learn/JavaScript/Asynchronous/Concepts#javascript_is_single_threaded">JavaScript is single-threaded</a>), llamado <b>main thread</b>.
+      </Text>
+
+      <Text fontSize={30}>
+        Todo lo que vimos hasta ahora es código sincrónico. Una sentencia se ejecuta una vez que la anterior finaliza.
+      </Text>
+
+      <Text fontSize={30}>
+        El código sincrónico nos impone algunas limitaciones cuando lo que debemos ejecutar es costoso, ya que JS no le devuelve el control al browser y éste puede parecer frizado, imposibilitando que maneje cualquier otra acción o tarea que el usuario solicite. A esto le llamamos código bloqueante.
```
Comment (nachosoloaga) [inline src/slides/class4/slides/BlockingCode.jsx:20]: Maybe `frizado` por `freezado`? Or another spanish word that represents what `freezed` means? 😄

File: src/slides/class4/slides/BlockingCode.jsx [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+      <iframe src={`${process.env.PUBLIC_URL}/snippets/class4/blockingCode.html`}></iframe>
+    </Slide>
+
+    <Slide backgroundColor="#0d0d0d">
+      <Heading margin="0px" fontSize="50px">
+        Web Workers
+      </Heading>
+
+      <Text fontSize={30}>
+        Los <a href="https://github.com/JulianPasquale/Webpack-templating/commit/2192a8391ccdb897ee806412a513227810131135">Web Workers</a> hacen posible la ejecución de código JS en background, en un hilo separado del main thread. De esta forma podemos ejecutar tareas laboriosas o costasas sin bloquear completamente el navegador.
```
Comment (nachosoloaga) [inline src/slides/class4/slides/BlockingCode.jsx:53]: `... en un hilo separado del main thread` por `... en un hilo separado del hilo principal` or maybe omitting the last `hilo`

Also, typo on `costasas`

File: src/slides/class4/slides/BlockingCode.jsx [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+      </Text>
+
+      <Text fontSize={30}>
+        Podemos utilizar el constructor <a href="https://developer.mozilla.org/en-US/docs/Web/API/Worker/Worker">Worker</a> para crear un worker a partir de un script. Además un worker puede invocar a otro worker.
+      </Text>
+
+      <Text fontSize={30}>
+        Los workers se comunican entre ellos y también con el main thread a través del pasaje de mensajes utilizando las directivas <a href="https://developer.mozilla.org/en-US/docs/Web/API/Worker/postMessage">Worker.postMessage</a> y <a href="https://developer.mozilla.org/en-US/docs/Web/API/Window/postMessage">Window.postMessage</a>
```
Comment (nachosoloaga) [inline src/slides/class4/slides/BlockingCode.jsx:61]: I think both directives are the same here

**Review by nachosoloaga — APPROVED** [review]

File: src/slides/class4/slides/BlockingCode.jsx (hunk 3 above)
Comment (JulianPasquale) [reply src/slides/class4/slides/BlockingCode.jsx:61]: Practically yes, they are. But the same directive is implemented in the Worker interface and the Window interface and they also differ in the arguments you can provide.

```javascript
// for window
window.postMessage(message, targetOrigin, [transfer]);

// for workers
worker.postMessage(message, [transfer]);
```

Comment (nachosoloaga) [reply src/slides/class4/slides/BlockingCode.jsx:61]: Oh I see, cool! I didn't realize they were invoked on different objects 😅

**Pull request merged**